sdk/synapse/synapse/src/models/index.ts

```diff
@@ -0,0 +1,970 @@
+/*
+ * Code generated by Microsoft (R) AutoRest Code Generator.
+ * Changes may cause incorrect behavior and will be lost if the code is regenerated.
+ */
+
+import { BaseResource, CloudError, AzureServiceClientOptions } from "@azure/ms-rest-azure-js";
+import * as msRest from "@azure/ms-rest-js";
+
+export { BaseResource, CloudError };
+
+/**
+ * An interface representing SparkJob.
+ */
+export interface SparkJob {
```

Comment:
> Are all of these fields truly optional? It seems like at least some should always be present? (e.g. name, state, submitTime)

They are optional, if the the `detailed` is false the submitTime won't be returned by service

Reply:
So if detailed isn't true, SparkJobs can be completely empty? I notice there are some monitoring APIs that seem to essentially just return a list of SparkJobs...so then with `detailed: false` those would only be useful in knowing the number of jobs?

Reply:
> So if detailed isn't true, SparkJobs can be completely empty? I notice there are some monitoring APIs that seem to essentially just return a list of SparkJobs...so then with `detailed: false` those would only be useful in knowing the number of jobs?

if detailed isn't true, it will only return id,state,appid,log,appinfo

Reply:
Can we mark the fields that are always returned as required then? (I'm fine if that ends up happening in the next update since this is in preview)

Reply:
This makes sense and benefits to our customers. We should do this in next version. I created a work item in our internal Azure Devops project and the GitHub issue [here](https://github.com/Azure/azure-rest-api-specs/issues/8530).